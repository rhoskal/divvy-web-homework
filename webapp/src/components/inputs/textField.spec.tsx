import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Form, Formik } from "formik";
import * as Yup from "yup";

import { TextField } from "./TextField";

describe("[Component] Input", () => {
  const onSubmit = jest.fn();

  it("should be disabled", () => {
    render(
      <Formik initialValues={{ email: "" }} onSubmit={onSubmit}>
        {(): React.ReactNode => (
          <Form>
            <TextField
              autoComplete="username"
              data-testid="username"
              disabled
              id="email"
              label="Email"
              name="email"
              type="email"
            />
          </Form>
        )}
      </Formik>,
    );

    const el = screen.getByTestId("username");

    expect(el).toHaveAttribute("disabled");
  });

  it("should handle validation errors", async () => {
    render(
      <Formik
        initialValues={{ email: "" }}
        onSubmit={onSubmit}
        validationSchema={Yup.object().shape({
          email: Yup.string().email("Please enter a valid email address").required(),
        })}>
        {(): React.ReactNode => (
          <Form>
            <TextField
              autoComplete="username"
              data-testid="username"
              id="email"
              label="Email"
              name="email"
            />
            <button type="submit">Submit</button>
          </Form>
        )}
      </Formik>,
    );

    const inputEl = screen.getByTestId("username");
    const btnEl = screen.getByText("Submit");

    userEvent.click(btnEl);

    await waitFor(() => {
      expect(inputEl).toHaveStyle({
        "border-color": "#f48fb1",
      });
    });
  });

  it("should match snapshot", () => {
    const { container } = render(
      <Formik initialValues={{ email: "" }} onSubmit={onSubmit}>
        {(): React.ReactNode => (
          <Form>
            <TextField autoComplete="username" id="email" label="Email" name="email" type="email" />
          </Form>
        )}
      </Formik>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
