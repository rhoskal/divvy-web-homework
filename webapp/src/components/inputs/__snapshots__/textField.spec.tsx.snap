// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Input should match snapshot 1`] = `
.emotion-0 {
  position: relative;
  margin-bottom: 16px;
}

.emotion-0 label {
  display: inline-block;
  color: #8d8d8d;
  font-size: 1rem;
  pointer-events: none;
  margin-bottom: 4px;
}

.emotion-0 input {
  padding: 14.5px 10px;
  font: inherit;
  color: currentColor;
  width: 100%;
  border-color: #d8d8d8;
  border-radius: 4px;
  border-style: solid;
  height: 1.1876em;
  margin: 0;
}

<form
  action="#"
>
  <div
    class="emotion-0"
  >
    <label
      for="email"
    >
      Email
    </label>
    <input
      autocomplete="username"
      name="email"
      type="email"
      value=""
    />
  </div>
</form>
`;
